lib/collection/src/shards/shard_holder.rs

```diff
@@ -49,10 +49,32 @@ pub struct ShardHolder {
 
 pub type LockedShardHolder = RwLock<ShardHolder>;
 
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+pub enum RingsKey {
+    Default,
+    ShardKey(ShardKey),
+    Resharding,
+}
```

Comment:
Having a look at this again, I don't like this design.

Shards under a shard key are completely isolated. They could be doing resharding at the same time. This design does not support that. We could extend it with more variants to make that happen, but that would not be a nice solution.

I'll try to submit a different approach in a separate PR so we can discuss.

Reply:
I propose we merge this and refactor it later. It's just quick-and-dirty PR that adds these types in the most trivial way, so we can start iterating and testing more important stuff. :/

Reply:
Sure, lets do that!

Reply:
Thanks! 🙏

I agree that this design is not adequate as a final solution, but it just gets the ball rolling. IMO, it feels like refactoring later would take same (or less) amount of time than trying to design it properly now.